cupyx/scipy/special/_bessel.py

```diff
@@ -52,7 +71,9 @@
 # Note: oddly, unlike for y0 or y1, SciPy always returns double for yn
 # dd->d because SciPy will accept 2.0
 yn = _core.create_ufunc(
-    'cupyx_scipy_special_yn', ('id->d', 'dd->d'),
+    'cupyx_scipy_special_yn',
+    (('ff->f', 'out0 = ynf((int)in0, in1)'),
```

Comment:
How about adding `if->f` case to utilize this optimization when the first argument is an int? And also keep `id->d`?

Reply:
I tried to add more signatures, but adding `if->f` for example, breaks tests: for `intc` order and `float16` array type cupy returns `float32`, when scipy retruns `float64`. I tried to change signatures order, add separate signatures for float16, but could not achieve both green tests and performance speedup. Maybe I dont understand any basic concept about ufuncs, but current implementation works, all tests are passed, it seems like everything works fast enough at least for float32 and float64 and you already spent more than enough time on this PR

Benchmark results (gpu time only):
| order type | arr type | gpu time   |
|------------|----------|------------|
| int        | float32  | 61.669 us  |
| int        | float64  | 301.463 us 
| float32    | float32  | 59.674 us  ||
| float32    | float64  | 299.772 us |


Bench src:
```
import cupy as cp
import cupyx
import itertools

x = cp.linspace(0, 4, 40000)
order = 1.1
combinations = itertools.product(
    (cp.intc(order), cp.float32(order), cp.float64(order)),
    (x.astype(cp.float32), x.astype(cp.float64))
)

for args in combinations:
    res_type = cupyx.scipy.special.yn(*args)
    res = cupyx.profiler.benchmark(cupyx.scipy.special.yn, args)
    print(f"{type(args[0].dtype)} | {args[1].dtype} -> {res_type.dtype} \n {res} \n")